plugins/outputs/azure_iothub/iothub/iotdevice/transport/transport.go

```diff
@@ -0,0 +1,52 @@
+package transport
+
+import (
+	"context"
+	"crypto/tls"
+	"github.com/amenzhinsky/iothub/logger"
+	"time"
+
+	"github.com/amenzhinsky/iothub/common"
+)
+
+// Transport interface.
+type Transport interface {
```

Comment:
This interface is implemented by the ./mqtt/mqtt.go with module-specific client and functions in ./mqtt/module_mqtt.go and this establishes a connection with the edgeHub/iotedge runtime. The edgeHub then connects to the Azure IoT Hub service via AMQP 1.0 in the scenario tested.

Next step would be to include an AMQP connection to the edgeHub. It would need to conform to the interface here. 

Note: Some of these functions can't be implemented for a module connection. For example, direct methods don't currently work for edge devices/modules. Any errors when calling this on an edge client are expected.